### PR TITLE
[*] Chore: Fix all new lint warnings, plus manual exports clean-up

### PR DESCRIPTION
**Pull request by etrepum**
## Description

* Runs @lexical/eslint-plugin over the source to resolve all new warnings
* Manually fix up names that were shadowed
* Manually fix up invariants, flow types, and docs for renamed and deprecated public API
* Manually fix up internal usage of newly deprecated APIs

Newly deprecated APIs (exports still available):

* @lexical/link toggleLink - renamed to $toggleLink
* @lexical/offset createChildrenArray - renamed to $createChildrenArray
* @lexical/selection trimTextContentFromAnchor - renamed to $trimTextContentFromAnchor

## Test Plan

* Manually verify that the public API is stable in code review, either through no change or through a @deprecated alias
* Manually verify that the flow types for any renamed or added public APIs are changed accordingly
* All tests still pass

## Review notes:

This is a really big PR if you read through the automated stuff. It could be reviewed in pieces:

* [Configuration change](https://github.com/facebook/lexical/pull/5979/commits/bac16e6face61d40be785b3447fe5757d697a47f) (first commit) - updateEditor is basically editor.update and should be considered a lexical provider
* [Automated `npm run lint -- --fix`](https://github.com/facebook/lexical/pull/5979/commits/1498f7db134672dae265a6cb780cc39db8a05ed5) (second commit) - this is completely automated and probably does not need close review
* [Manual fix-ups](https://github.com/facebook/lexical/pull/5979/files/1498f7db134672dae265a6cb780cc39db8a05ed5..1c60e7bf04cd06e8d465c460f34105881e632e0d) (almost everything after the second commit) - worth a close look
* [Doc improvements](https://github.com/facebook/lexical/pull/5979/commits/d5b6ea8f01aa229b1d63ef20049da423f56ac641) (last relevant commit) - adds `{@link $newName}` around deprecations for better typedoc ux, and removes unused markdown partial plugin
* [Flaky collab test mitigation?](https://github.com/facebook/lexical/pull/5979/commits/58a94e6e8d59343d194bf48d794558a5c6b8ae01) (not so relevant commit) adds some stuff to maybe help address collab tests that seem flaky. It seemed to help a bit locally, hard to tell whether it it's an solid improvement in CI or not until a lot of test runs have happened.

### CODE REVIEW

File: packages/lexical-yjs/src/Utils.ts [hunk 1]
```diff
@@ -476,26 +472,6 @@ export function syncWithTransaction(binding: Binding, fn: () => void): void {
...
-      invariant(false, 'createChildrenArray: node does not exist in nodeMap');
-    }
-    children.push(nodeKey);
-    nodeKey = node.__next;
-  }
-  return children;
-}
-/** @deprecated renamed to $createChildrenArray by @lexical/eslint-plugin rules-of-lexical */
-export const createChildrenArray = $createChildrenArray;
-
```
Comment (etrepum) [inline packages/lexical-yjs/src/Utils.ts:493]: This is a dupe of the same function from @lexical/offset which is already a dependency of @lexical/yjs

**Comment by etrepum** [comment]
Pretty sure the test failure is just a flaky test, I can't repro locally on my mac and the tests succeeded earlier

`CI=true npm run test-e2e-collab-ci-firefox`

```
[1]   1) [firefox] › packages/lexical-playground/__tests__/regression/4697-repeated-table-selection.spec.mjs:21:3 › Regression test #4697 › repeated table selection results in table selection 
[1] 
[1]     Error: expect(received).toEqual(expected) // deep equality
[1] 
[1]     Expected: 1
[1]     Received: 2
[1] 
[1]        at packages/lexical-playground/__tests__/utils/index.mjs:267
[1] 
[1]       265 |     }
[1]       266 |     if (coordinates.anchor.y !== undefined) {
[1]     > 267 |       expect(_anchor.y).toEqual(coordinates.anchor.y);
[1]           |                         ^
[1]       268 |     }
[1]       269 |   }
[1]       270 |   if (coordinates.focus) {
[1] 
[1]         at assertTableSelectionCoordinates (/Users/bob/src/lexical/packages/lexical-playground/__tests__/utils/index.mjs:267:25)
[1]         at /Users/bob/src/lexical/packages/lexical-playground/__tests__/regression/4697-repeated-table-selection.spec.mjs:49:5
[1] 
[1]   Slow test file: [firefox] › packages/lexical-playground/__tests__/e2e/Markdown.spec.mjs (1.6m)
[1]   Slow test file: [firefox] › packages/lexical-playground/__tests__/e2e/Tables.spec.mjs (1.5m)
[1]   Slow test file: [firefox] › packages/lexical-playground/__tests__/e2e/List.spec.mjs (1.2m)
[1]   Slow test file: [firefox] › packages/lexical-playground/__tests__/e2e/Links.spec.mjs (1.1m)
[1]   Slow test file: [firefox] › packages/lexical-playground/__tests__/e2e/Navigation.spec.mjs (59.1s)
[1]   Consider splitting slow test files to speed up parallel execution
[1]   1 flaky
[1]     [firefox] › packages/lexical-playground/__tests__/regression/4697-repeated-table-selection.spec.mjs:21:3 › Regression test #4697 › repeated table selection results in table selection 
[1]   79 skipped
[1]   362 passed (4.1m)
[1] npm run test-e2e-collab-firefox exited with code 0
```

**Comment by ivailop7** [comment]
@etrepum ok, I'm ready to get this one through. Rebase to latest main and I'll approve and merge it. Hope your 'ONCL' offer still stands.

**Comment by etrepum** [comment]
@ivailop7 as long as it happens in the morning, afternoon is no longer available. Just rebased.

**Comment by ivailop7** [comment]
Which one should be the go to PR, this one or the #6078

**Comment by etrepum** [comment]
@ivailop7 they are the same but I don't know why the checks are not running and created another PR to try and figure it out. They are still not running, I don't know what to do here.

**Comment by etrepum** [comment]
@ivailop7 I don't think it's related to this changeset because it doesn't touch workflows and other PRs from today are affected (#6076, #6077 at least)

**Comment by etrepum** [comment]
Unit test suites are passing locally

```
$ git rev-parse HEAD
58a94e6e8d59343d194bf48d794558a5c6b8ae01
$ npm run test-unit
[…]
Test Suites: 50 passed, 50 total
Tests:       2 skipped, 1367 passed, 1369 total
Snapshots:   0 total
Time:        86.149 s
Ran all test suites.
```

**Comment by etrepum** [comment]
e2e suite runs with mac+chromium

```
$ npm run clean && CI=true npm run test-e2e-ci-chromium
  […]
  1 flaky
    [chromium] › packages/lexical-playground/__tests__/e2e/Links.spec.mjs:1959:3 › Links › Can handle pressing Enter inside a Link containing multiple TextNodes 
  19 skipped
  428 passed (2.0m)
```

I'll do a prod collab suite next but given the nature of these changes I don't expect that a rename could make a difference to any of that.

**Comment by etrepum** [comment]
@ivailop7 mac collab prod chromium is good to go as well, I have full confidence that there are no new regressions in this rebase.

```
$ npm run clean && CI=true npm run test-e2e-collab-prod-ci-chromium
[…]
[1]   2 flaky
[1]     [chromium] › packages/lexical-playground/__tests__/e2e/Links.spec.mjs:1959:3 › Links › Can handle pressing Enter inside a Link containing multiple TextNodes 
[1]     [chromium] › packages/lexical-playground/__tests__/e2e/Tables.spec.mjs:1267:3 › Tables › Grid selection: can select multiple cells and insert an image 
[1]   65 skipped
[1]   381 passed (2.8m)
```

**Comment by etrepum** [comment]
ah also ci-check passes

```
$ npm run ci-check
[…]
$ echo $?
0
```

**Comment by ivailop7** [comment]
@etrepum ok, game time

**Review by ivailop7 — APPROVED** [review]

**Pull request merged**